### PR TITLE
Implemented functions

### PR DESCRIPTION
**Pull request by Lynette7**

### CODE REVIEW

File: grammer.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
 <term> ::= <factor> | <term> <mulop> <factor>
 <factor> ::= <number> | '(' <expression> ')' | <addop> <factor> | <identifier> | <string> | <atom>
 <atom> ::= <if-expression> | <while-expression> | <for-expression>
+<function-definition> ::= 'func' <identifier>? '(' (<identifier> [',' <identifier>]*)? ')' '-' '>' <expression>
```
Comment (RyanKoech) [inline grammer.html:10]: Shouldn't out grammar for functions be like `func(<zero or more identifiers seperated by commas>) { <statement> }`? Also the `+ ?` and `()` are usually not found in BNFs right?

**Review by RyanKoech — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -284,6 +285,17 @@ def make_and(self):
         return None, ExpectedCharError(pos_start, self.pos, "'&' (after '&')")
         
     
+    def make_minus_or_arrow(self):
```
Comment (RyanKoech) [inline main.py:288]: I don't think we'll need this anymore.

File: main.py [hunk 3]
```diff
@@ -156,8 +158,7 @@ def make_tokens(self):
                 tokens.append(Token(TOKEN_PLUS, pos_start=self.pos))
                 self.advance()
             elif self.current_char == '-':
-                tokens.append(Token(TOKEN_MINUS, pos_start=self.pos))
-                self.advance()
+                tokens.append(self.make_minus_or_arrow())
```
Comment (RyanKoech) [inline main.py:161]: I don't think we need this anymore.

**Review by RyanKoech — CHANGES_REQUESTED** [review]
Finally, I think you need to implement the parser to accept function calls.

**Review by RyanKoech — APPROVED** [review]

**Review by m-vely — APPROVED** [review]

**Review by MainaKarl — APPROVED** [review]

**Pull request merged**